src/main/java/com/kodilla/ecommercee/controller/ProductController.java

```diff
@@ -0,0 +1,45 @@
+package com.kodilla.ecommercee.controller;
+
+import com.kodilla.ecommercee.controller.exceptions.ProductNotFoundException;
+import com.kodilla.ecommercee.domain.Product;
+import com.kodilla.ecommercee.repository.ProductRepository;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@CrossOrigin("*")
+@RequestMapping(value = "v1/product", produces = "application/json")
+public class ProductController {
+    private ProductRepository productRepository;
+
+    public ProductController(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    @GetMapping(value = "getProduct")
+    public Product getProduct(@RequestParam Long id) throws ProductNotFoundException {
+        return productRepository.findById(id).orElseThrow(ProductNotFoundException::new);
+    }
+
+    @GetMapping(value = "getProducts")
+    public List<Product> getProducts() {
+        return productRepository.findAll();
+    }
+
+    @PostMapping(value = "addProduct", consumes = "application/json")
+    public void addProduct(@RequestBody Product product) {
+        productRepository.save(product);
+    }
+
+    @PutMapping(value = "updateProduct", consumes = "application/json")
```

Comment:
ten `consumes = "application/json"` jest potrzebny jeśli mamy @RestController ?

Reply:
Generalnie z tego co kiedyś sprawdzałem to w przypadku gdy tego nie wpiszemy nie jest wymagany taki format danych przez springa